src/cozy/wrappers/sharing/sharing.component.ts

```diff
@@ -160,4 +170,19 @@ export class SharingComponent extends AngularWrapperComponent {
             rejectRecipient: rejectUser,
         };
     }
+
+    /*********************/
+    /* Sharing Listeners */
+    /*********************/
+
+    protected async onShare({
+        document,
+        recipients,
+        readOnlyRecipients,
+    }: OnShareEventArgs) {
+        const recipientsToConfirm = [...recipients, ...readOnlyRecipients];
+        const organizationId = document.id;
+
+        this.sharingService.autoConfirmTrustedUsers(organizationId, recipientsToConfirm);

```

Comment:
no need to await?

Reply:
Not necessary as it is just a callback but you're right. I added the `await` and I changed cozy-sharing to await the callback to finish (PR not created yet)